### PR TITLE
Add support for base_url in subscriptions client

### LINKED ISSUE #705: add base_url to subscriptions client
Add base_url support to subscriptions client.

in client, construct urls from a base_url pulled from a global (which in turn is constructed from the planet.base_url constant).

NOTE: be sure to remove `root_url` variables in the subscription client code.  ([discussion](https://github.com/planetlabs/planet-client-python/pull/701#discussion_r986194642_))

This resolves some of #628

### PR DESCRIPTION
**Pull request by jreiberkyle**
**Related Issue(s):**

Closes #705

**Proposed Changes:**

For inclusion in changelog (if applicable):

1. Add support for base_url in subscriptions client

Not intended for changelog:

1. 

**Diff of User Interface**

Old behavior:

Not possible to specify base_url.

New behavior:

can specify base_url with e.g. `planet subscriptions --base-url <BASE_URL> list` and 

```python
async with Session() as sess:
    cl = SubscriptionsClient(sess, base_url=<BASE_URL>)
    ...
```  

**PR Checklist:**

- [x] This PR is as small and focused as possible
- [x] If this PR includes proposed changes for inclusion in the changelog, the title of this PR summarizes those changes and is ready for inclusion in the Changelog.
- [x] I have updated docstrings for function changes and docs in the 'docs' folder for user interface / behavior changes
- [x] This PR does not break any examples or I have updated them

**(Optional) @mentions for Notifications:**

### CODE REVIEW

**Comment by jreiberkyle** [comment]
Initial failure of coverage check likely due to #723. Re-ran and now it passed

File: planet/clients/subscriptions.py [hunk 1]
```diff
@@ -6,6 +6,9 @@
 from planet.exceptions import APIError, ClientError
 from planet.http import Session
 from planet.models import Paged
+from ..constants import PLANET_BASE_URL
+
+BASE_URL = f'{PLANET_BASE_URL}/subscriptions/v1/'
```
Comment (kevinlacaille) [inline planet/clients/subscriptions.py:11]: Thank you for moving this base URL here ([related thread](https://github.com/planetlabs/planet-client-python/pull/701#discussion_r986184740)). 👍

File: planet/cli/subscriptions.py [hunk 2]
```diff
@@ -10,11 +10,22 @@
 from planet.clients.subscriptions import SubscriptionsClient
 
 
+@asynccontextmanager
+async def subscriptions_client(ctx):
```
Comment (kevinlacaille) [inline planet/cli/subscriptions.py:14]: Nice! Great call moving the CLI session here 🥇

File: planet/clients/subscriptions.py [hunk 3]
```diff
@@ -22,15 +25,23 @@ class SubscriptionsClient:
...
     """
 
-    def __init__(self, session: Session) -> None:
+    def __init__(self,
+                 session: Session,
+                 base_url: Optional[str] = None) -> None:
+        """
+        Parameters:
+            session: Open session connected to server.
+            base_url: The base URL to use. Defaults to production subscriptions
```
Comment (kevinlacaille) [inline planet/clients/subscriptions.py:36]: I'm not sure I understand this client well enough, but what's the purpose of defining a base URL aside from the subscriptions API base URL?

File: tests/integration/test_subscriptions_cli.py [hunk 4]
```diff
@@ -58,17 +69,16 @@ def test_subscriptions_list_options(options, expected_count):
 
 
 @failing_api_mock
-def test_subscriptions_create_failure():
+def test_subscriptions_create_failure(invoke):
     """An invalid subscription request fails to create a new subscription."""
     # This subscription request lacks the required "delivery" and
     # "source" members.
     sub = {'name': 'lol'}
```
Comment (kevinlacaille) [inline tests/integration/test_subscriptions_cli.py:76]: love this - `name: lol`

File: tests/integration/test_subscriptions_cli.py [hunk 5]
```diff
@@ -36,20 +36,31 @@
 # CliRunner().invoke(cli.main, args=['subscriptions', 'list', limit]
 #
 # does not work.
+
+
+@pytest.fixture
+def invoke():
```
Comment (kevinlacaille) [inline tests/integration/test_subscriptions_cli.py:42]: Nice call placing this in an `invoke` function, like we do for the Data and Orders CLI tests 👍

**Review by kevinlacaille — APPROVED** [review]
Lots of small and very sensible changes! Also, kudos on managing on neither adding nor removing a single line of code to the codebase 🤯

File: planet/clients/subscriptions.py (hunk 3 above)
Comment (jreiberkyle) [reply planet/clients/subscriptions.py:36]: Redirecting to a test service is the primary use case. @cholmes has been using it for that. Also, QA team could redirect to a QA service.

File: planet/clients/subscriptions.py (hunk 1 above)
Comment (jreiberkyle) [reply planet/clients/subscriptions.py:11]: great memory! Yep, this has been on my mind since that discussion and I'm glad to have it resolved.

**Pull request merged**